great_expectations/experimental/datasources/interfaces.py

```diff
@@ -311,8 +310,14 @@ def sort_batches(self, batch_list: List[Batch]) -> None:
 DataAssetType = TypeVar("DataAssetType", bound=DataAsset)
 
 
+# It would be best to bind this to ExecutionEngine, but we can't now due to circular imports
+ExecutionEngineType = TypeVar("ExecutionEngineType")
```

Comment:
You don't need to actually do this but...
We could bind it to an `ExecutionEngine` protocol that defines whatever methods we actually call within `zep` code.

Reply:
Also not necessarily needed in this PR but I think we should change how we name our `TypeVar` so that they are more obviously `TypeVar`s.

Here's the same issue being discussed in `typeshed`
https://github.com/python/typeshed/issues/1872

Reply:
After internal discussion name was changed to `_ExecutionEngineT`.